### PR TITLE
Diagnose and Resolve Potential Run Failure Registration

### PR DESCRIPTION
**Pull request by Tohron**
This PR is to assist in ensuring that run failures get registered properly.  It currently adds logging to the run failure detection system and resolves a potential issue in benchmarks_ci.py.

### CODE REVIEW

**Comment by DrewScoggins** [comment]
So I took a look at a recent failure and this is what was in the log. It seems like FailureReporting.exe does not exist where we expect it to.

```
[2021/04/19 05:03:51][INFO] $ pushd "/home/helixbot/work/B9B30A1C/p/FailureReporter"
[2021/04/19 05:03:51][INFO] $ FailureReporting.exe traces/FailureReporter/failure-report.json
[2021/04/19 05:03:51][INFO] $ popd
Traceback (most recent call last):
  File "/home/helixbot/work/B9B30A1C/p/shared/startup.py", line 114, in runtests
    RunCommand(startup_args, verbose=True).run()
  File "/home/helixbot/work/B9B30A1C/p/scripts/performance/common.py", line 206, in run
    returncode, quoted_cmdline)
subprocess.CalledProcessError: Command '$ /home/helixbot/work/B9B30A1C/p/startup/Startup --app-exe pub/emptycsconsoletemplate --metric-type TimeToMain --trace-name "Startup - New Console Template - FDD Publish_startup" --gui-app false --process-will-exit true --iterations 5 --timeout 50 --warmup true --working-dir /home/helixbot/work/B9B30A1C/w/BA900ABF/e --report-json-path traces/perf-lab-report.json --trace-directory traces --scenario-name "Startup - New Console Template - FDD Publish" --environment-variables COMPlus_EnableEventLog=1' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 13, in <module>
    Runner(traits).run()
  File "/home/helixbot/work/B9B30A1C/p/shared/runner.py", line 185, in run
    startup.runtests(self.traits)
  File "/home/helixbot/work/B9B30A1C/p/shared/startup.py", line 128, in runtests
    RunCommand(cmdline, verbose=True).run(reporterpath)
  File "/home/helixbot/work/B9B30A1C/p/scripts/performance/common.py", line 197, in run
    (returncode, quoted_cmdline) = self.__runinternal(working_directory)
  File "/home/helixbot/work/B9B30A1C/p/scripts/performance/common.py", line 182, in __runinternal
    universal_newlines=True,
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'FailureReporting.exe': 'FailureReporting.exe'
```

**Comment by Tohron** [comment]
Yeah, I noticed that earlier, and asked whether -o $(CorrelationStaging)FailureReporter in scenarios.yml would put the executable where benchmarks_ci.py expects it to be.  I don't remember getting an answer.

File: src/scenarios/shared/startup.py [hunk 1]
```diff
@@ -113,18 +114,24 @@ def runtests(self, traits: TestTraits):
...
             if runninginlab():
                 upload_container = 'failedresults'
                 reportjson = os.path.join(
                     TRACEDIR,
                     'FailureReporter', 
                     'failure-report.json')
+                if iswin:
+                    executable = 'FailureReporting.exe'
+                else:
+                    executable = 'FailureReporting'
```
Comment (billwert) [inline src/scenarios/shared/startup.py:127]: Take a look at this example:
https://github.com/dotnet/performance/blob/0c988339004e22574f83c36ea12179a1c26aafc0/src/scenarios/shared/startup.py#L54

I think we should do something similar here. You can port `extension()` as well.

**Review by billwert — CHANGES_REQUESTED** [review]
See note in `startup.py`. It also should apply to `benchmarks_ci.py`.

**Review by billwert — APPROVED** [review]

**Pull request merged**